### DISCUSSION TITLE
Building wheel with mypyc

### DISCUSSION THREAD

**Discussion by char101** [jsh9/cercis#34] [General]
Hi,

Just want to share the steps to build the wheel with mypyc, since it is not obvious how to do it. This is done in Windows environment.

Install build requirements:
```batch
pip install mypy hatch hatch-mypyc hatch-fancy-pypi-readme
```

Enable the mypyc hook:
```batch
set HATCH_BUILD_HOOK_ENABLE_MYPYC=true
```

Build the wheel:
```batch
hatch build -t wheel
```

**Comment by jsh9** [comment]
Hi @char101 , I'm a bit confused by your post.  Is this an instruction of building _Cercis_?  Thanks!

**Comment by char101** [comment]
Yeah, if you have noticed black is uploaded to pypi as a native wheel compiled with mypyc which results in 2x faster performance. Since cercis is a fork of black it should be able to be compiled the same way.

**Reply by jsh9** [reply]
Does it mean that I need to tweak this config file then?  https://github.com/jsh9/cercis/blob/23420e6f14ba4b979e128d80942c15243c1a19b7/.github/workflows/python-publish.yml

Thanks!

**Reply by char101** [reply]
Sorry I'm not familiar with the workflow. It works fine without compiling with mypyc so I don't think it is required to also uploads the mypyc versions.
